src/event/reader.rs

```diff
@@ -107,6 +109,29 @@ pub struct EventReader {
     rg_state: Arc<Mutex<ReaderGroupState>>,
 }
 
+impl Drop for EventReader {
+    /// Destructor for reader invoked. This will automatically invoke reader_offline().
+    fn drop(&mut self) {
+        info!("reader {} is dropped", self.id);
+        // try fetching the currently running Runtme.
+        let r = Handle::try_current();
+        match r {
+            Ok(handle) => {
+                // enter the runtime context.
+                let _ = handle.enter();
+                // ensure we block until the reader_offline method completes.
+                futures::executor::block_on(self.reader_offline());
```

Comment:
I check around and see people are saying that this way is not encouraged. I guess we use this workaround for now and come back if better solution comes up.